pkg/util/system/network_linux.go

```diff
@@ -0,0 +1,158 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package system
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/filesystem"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// NetworkRoute holds one network destination subnet and it's linked interface name
+type NetworkRoute struct {
+	Interface string
+	Subnet    uint64
+	Gateway   uint64
+	Mask      uint64
+}
+
+// ParseProcessRoutes parses /proc/<pid>/net/route into a list of NetworkDestionation
+// If PID is 0, it parses /proc/net/route instead
+func ParseProcessRoutes(procPath string, pid int) ([]NetworkRoute, error) {
+	var procNetFile string
+	if pid > 0 {
+		procNetFile = filepath.Join(procPath, strconv.Itoa(pid), "net", "route")
+	} else {
+		procNetFile = filepath.Join(procPath, "net", "route")
+	}
+
+	if !filesystem.FileExists(procNetFile) {
+		return nil, fmt.Errorf("%s not found", procNetFile)
+	}
+	lines, err := filesystem.ReadLines(procNetFile)
+	if err != nil {
+		return nil, err
+	}
+	if len(lines) < 1 {
+		return nil, fmt.Errorf("empty network file %s", procNetFile)
+	}
+
+	routes := make([]NetworkRoute, 0, len(lines))
+	for _, line := range lines[1:] {
+		fields := strings.Fields(line)
+		if len(fields) < 8 {
+			continue
+		}
+		dest, err := strconv.ParseUint(fields[1], 16, 32)
+		if err != nil {
+			log.Debugf("Cannot parse destination %q: %v", fields[1], err)
+			continue
+		}
+		gateway, err := strconv.ParseUint(fields[2], 16, 32)
+		if err != nil {
+			log.Debugf("Cannot parse gateway %q: %v", fields[2], err)
+			continue
+		}
+		mask, err := strconv.ParseUint(fields[7], 16, 32)
+		if err != nil {
+			log.Debugf("Cannot parse mask %q: %v", fields[7], err)
+			continue
+		}
+		d := NetworkRoute{
+			Interface: fields[0],
+			Subnet:    dest,
+			Gateway:   gateway,
+			Mask:      mask,
+		}
+		routes = append(routes, d)
+	}
+	return routes, nil
+}
+
+// // defaultGateway returns the default Docker gateway.
+// func defaultGateway(procPath string) (net.IP, error) {
```

Comment:
Why so much commented-out code?  Can this be put on a branch somewhere, for when it's needed?

Reply:
So this PR is migrating the `docker` check but it's only the first in the series (to migrate everything to the new generic metrics provider).
I keep that here as legacy code I moved because I am not sure if we will be able to get rid of it or not, it will be a bit easier not to have to find them again, so if it's that's ok I'd like to keep it there temporarily.

Reply:
GIt is _pretty_ good at keeping code around on its own!  In general, I think that's preferred, as commented-out code can quickly bitrot, and isn't tested.  Maybe a comment to say "defaultGateway was removed; see <link to tree before it was removed>"?  That puts the function in its context, so that if it _does_ bitrot, it's easier to see where that bitrotting might have occurred (`git log`, `git blame`, etc.).